lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -169,18 +171,19 @@ public void stop(long millisToWait) throws InterruptedException {
         }
     }
     
-    private enum ErrorType {
+    // error disposition
+    private enum ErrorDisp {
```

Comment:
How about consolidating all errors into one big enum,  or even better, externalize the errors in a properties file and read them into the enum or class. This will make it easier to add to documentation, and the ops folks can access easily.

Reply:
This enum and the ErrorType enum are entirely orthogonal - as I noted above, one is just a ternary selector for a log prefix. The new ErrorType enum actually contains error types.

For the ErrorType enum, it's not clear to me how loading the contents from a file would work, since the code selects specific known error codes based on various scenarios. How is would pick an error code from an arbitrary list of codes isn't clear.

In any case, IMO there are much larger priorities and we should wait for a request from ops before we implement something they may or may not want.

Reply:
Something like,
```
LOCATION_ERROR=Lack of location data for a location transform.
SUBOBJECT_COUNT=An object has too many subobjects to index. 
...
```
in a properties file that can be referenced from an ops manual in their entirety. 
Then, 
```
Enum ErrorTypes {
   properties = new Properties();

...
   try {
           properties.load(Constants.class.getResourceAsStream(PATH));
   }
...

public void getMessage(String type) {
    properties.get(type);  // returns message or throws error if type does not exist.
}
}
```

Reply:
But what is the point of that string? It wouldn't be used anywhere.

Reply:
Serves as a key/code/reference. Makes it easy when talking about the error.

Reply:
But if it's never used in the code, it won't serve that function. I'm only planning to stick the enum values into the DB, so any strings associated with those values will never see the light of day.

Reply:
OK. I was thinking along the lines of having error codes that could be used in log messages and exceptions with which logs can be easily searched etc. since you had once mentioned that you wanted that feature. But this is fine.

Reply:
No, the point of all this work is to *avoid* having to go to the logs. Hopefully once it's done the logs will only be needed for getting the entire stack trace for truncated traces (which will be easy to find by grepping for event ID) and for looking at retry rates, since the retries obviously won't be going in the DB. I don't see the latter happening very often.